build/compat.m4

```diff
@@ -0,0 +1,18 @@
+dnl taken from Autoconf's m4sh.m4, GPLv3+
+m4_ifndef([AS_VAR_COPY], [
+m4_define([AS_VAR_COPY],
+[AS_LITERAL_WORD_IF([$1[]$2], [$1=$$2], [eval $1=\$$2])])
+])
```

Comment:
I'm not totally sure about this part: it might not be needed, and if it is I'm not sure `AS_LITERAL_WORD_IF` would be available.  This code is what Autoconf has nowadays, but the inspiration I got from cryptsetup has `AS_LITERAL_IF` instead (but it seems to be buggy in some cases if I trust the documentation).
Anyway, we can fix this if a problem arises